x-pack/legacy/plugins/integrations_manager/public/screens/home/header.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  // @ts-ignore (elastic/eui#1557) & (elastic/eui#1262) EuiImage is not exported yet
+  EuiImage,
+  EuiPage,
+  EuiPageBody,
+  EuiPageWidthProps,
+  // @ts-ignore
+  EuiSearchBar,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import styled from 'styled-components';
+import { useCore, useLinks } from '../../hooks';
+
+export type HeaderProps = EuiPageWidthProps & {
+  onSearch: (userInput: string) => unknown;
+};
+
+const Page = styled(EuiPage)`
+  padding: 0;
+`;
+
+export function Header({ restrictWidth, onSearch }: HeaderProps) {
+  const [searchTerm, setSearchTerm] = useState('');
+  const searchBar = (
+    <EuiSearchBar
+      query={searchTerm}
+      key="search-input"
+      box={{
+        placeholder: 'Find a new package, or one you already use.',
+        incremental: true,
+      }}
+      onChange={({ queryText: userInput }: { queryText: string }) => {
+        setSearchTerm(userInput);
+        onSearch(userInput);
+      }}
+    />
+  );
+
+  const left = (
+    <EuiFlexGroup alignItems="center" gutterSize="none">
+      <EuiFlexItem>
+        <HeroCopy />
+        <EuiSpacer size="xl" />
+        <EuiFlexGroup>
+          <EuiFlexItem grow={4}>{searchBar}</EuiFlexItem>
+          <EuiFlexItem grow={2} />
+        </EuiFlexGroup>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+  const right = <HeroImage />;
+
+  return (
+    <Page>
+      <EuiPageBody restrictWidth={restrictWidth}>
+        <EuiFlexGroup gutterSize="none">
+          <EuiFlexItem grow={1}>{left}</EuiFlexItem>
+          <EuiFlexItem grow={1}>{right}</EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiPageBody>
+    </Page>
+  );
+}
+
+function HeroCopy() {
+  const { theme } = useCore();
+  const Subtitle = styled(EuiText)`
+    color: ${theme.eui.euiColorDarkShade};
+  `;
+
+  return (
+    <Fragment>
+      <EuiTitle size="l">
+        <h1>Add Your Data</h1>
+      </EuiTitle>
+      <Subtitle>Some creative copy about integrations goes here.</Subtitle>
```

Comment:
Is there something like a `TODO` we could add here that we don't miss to change it later?

Reply:
Great idea. I'll create a ticket and link in a comment here.

I read the highlighted code two or three times wondering what you were referencing before I got it. Which is exactly why we need a reminder 😉

Reply:
@ruflin 
> I tried this locally and when I start typing anything, it filters out all integrations. So when I type a or A nothing shows up. Did not investigate why.

We dug into this and found the issue was that the app was requesting `/api` instead of `/$BASE_PATH/api`. I changed the way data fetches happen & this is no longer an issue.

> Is it expected that every time the search term is deleted, it refetches (probably only internally) all the integrations?

Is was and I figured we'd get back to when applying caching logic, but my data fetching changes fixed this too. Have a look

> One thing I noticed which is not related to this PR is that the internal search API endpoint is called list (like it was on the registry API). It probably makes sense to rename it for consitency at one point.

Created https://github.com/elastic/kibana/issues/46854